lib/transfer.cpp

```diff
@@ -79,12 +79,27 @@ namespace quda {
       total_block_size *= geo_bs[d];
     }
 
-    if (total_block_size == 1) errorQuda("Total geometric block size is 1");
+    // Aggregation size is "technically" 1 for optimized KD
+    if (total_block_size != 1 && transfer_type == QUDA_TRANSFER_OPTIMIZED_KD)
+      errorQuda("Total geometric block size must be 1 for transfer type optimized-kd");
+    if (transfer_type == QUDA_TRANSFER_OPTIMIZED_KD) {
+      if (total_block_size != 1)
+        errorQuda("Invalid geometric block size %d for optimized-kd aggregation, must be 1", total_block_size);
```

Comment:
Looks like you're catching this error (`total_block_size != 1`) twice here

Reply:
Good catch, fixed